src/App.tsx

```diff
@@ -1,27 +1,73 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  type FilterParam = 'all-button' | 'first-five-button' | 'red-button';
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const setFilterParam = async (param: FilterParam) => {
+    let filteredArray: Good[];
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    switch (param) {
+      case 'all-button':
```

Comment:
use enum instead of type FilterParam and you will be able to use it in cases

case FiltersEnum.All

Reply:
Done!